WIP: Add dplyr join content (#61)
bcgov/ds-intro-to-r-2-day: xx-joins.Rmd

---
title: "Joining Related Data"
teaching: 35
exercises: 10
questions:
- "How can I change a dataframe based on information in another dataframe?"
objectives:
- "To understand the concepts of relational data and be able to change one table based on another with dplyr's `xxx_join()` functions."
keypoints:
- "Use the `dplyr` package to join relational data."
- "Use `left_join()`, `full_join()`, `right_join()` or `inner_join()` to add data from one table to another."
- "Use `semi_join()` or `anti_join()` to filter a table based on data in another table."
source: Rmd
---

> # Relational Data

Reviewer: `# Joining Related Data`

> # Relational Data

Reviewer: ```suggestion
# Joining Related Data
```

```{r, include=FALSE}
source("bin/chunk-options.R")
knitr_fig_path("xx-")
```

> 

Reviewer: ## Relational Data

> 

Reviewer: ```suggestion
## Introducing Relational Data
```

> In real life, we often have two or more related datasets that have a key 
> linking the data.

Reviewer: ```suggestion
In real life, we often have two or more related datasets that share a set of common columns.   We would like to combine these two datasets based on their common columns, also called a **key** or **key variables**.
```

We may want to change a dataset by adding information from a second dataset.

> For example, one dataset may contain employee information such as employee ID, 
> name, job position, hire date, etc., whereas a second dataset may contain the
> current and historical wages for each job position. A relevant join -- adding a 

Reviewer: ```suggestion
For example, say we have a dataset containing employee information such as employee ID, 
name, job position, hire date, etc., and a second dataset containing the
current and historical wages for each job position. A relevant join -- adding a 
```

column of data for each row -- may be merging the current wage for each employee 
into the employee dataset.

> We may also want to choose only a portion of a dataset. For example, one dataset
> may include all students who ever attended a school along with their student ID, 
> years attended, grades for each class, graduation date (if applicable), and so on,
> whereas a second dataset may be simply a list of current students. A relevant join

Reviewer: ```suggestion
We may also want to choose only a portion of a dataset. For example, say we have a dataset including all students who ever attended a school along with their student ID, 
years attended, grades for each class, graduation date (if applicable), and so on,
and a second dataset which is simply a list of current students. A relevant join
```

-- retaining all columns of data, but for only specific rows -- may be a reduction 
of the first dataset for only those students currently in attendance, but with all 
their information (e.g., grades, etc.).

The first example is that of a **mutating join** - where we change one table 
by adding in data from a second table, based on matching through a key variable 
that exists in both tables. The second example is that of a **filtering join** - 
where we subset one table based on data in a second table, again based on matching 
through a key variable.

## Types of joins with dplyr

The `dplyr` package has several functions that allow us to make these joins. 
Assume we have tables _x_ and _y_.

There are four types of **mutating joins**, which add new variables to one 
table from matching rows in another:

> * **inner joins** include all rows in x AND y

Reviewer: I'd be tempted to expand the vignette descriptions to include the matching rows.  Something like:
**left_joins** include all rows in x *plus* any row in y with a match to a row in x
